app/requisiciones/routes.py

```
@@ -0,0 +1,683 @@
+        if not any(nuevo_estado == choice[0] for choice in opciones_estado_permitidas):
+            flash('Intento de cambio de estado no válido o no permitido para su rol/estado actual.', 'danger')
+            return redirect(url_for('requisiciones.ver_requisicion', requisicion_id=requisicion.id)) # Changed blueprint name
+
+        comentario_ingresado_texto = form_estado.comentario_estado.data.strip() if form_estado.comentario_estado.data else None
+
+        if requisicion.estado != nuevo_estado or (comentario_ingresado_texto and comentario_ingresado_texto != requisicion.comentario_estado):
+            if nuevo_estado in ['Rechazada por Almacén', 'Rechazada por Compras', 'Cancelada'] and not comentario_ingresado_texto:
+                flash('Es altamente recomendable ingresar un motivo al rechazar o cancelar la requisición.', 'warning')
+
+            from ..models import Rol
+            if utils.cambiar_estado_requisicion(
```

Pylint (3.3.7)

[error] Module 'app.utils' has no 'cambiar_estado_requisicion' member

(E1101)